server/auth/types/basic/routes.ts

```diff
@@ -55,24 +53,31 @@ export class BasicAuthRoutes {
       async (context, request, response) => {
         const forbiddenUsernames = this.config.auth.forbidden_usernames;
         if (forbiddenUsernames.indexOf(request.body.username) > -1) {
```

Comment:
What is the type of `forbiddenUsernames`? Array of string or single string like `user1,user2,user3`? If it is a single string, in this way it will also block the user like user123 since it has a sub-string of user1, right?

Reply:
it is array of strings, the config schema is defined in `server/index.ts`